File: clients/csharp/ActClient.cs
namespace Anticaptrad;
public sealed class ActClient(string baseUrl) { private readonly string _baseUrl = baseUrl.TrimEnd('/'); public string Url(string path) => _baseUrl + (path.StartsWith('/') ? path : "/" + path); public string HealthUrl => Url("/health"); public string ReadyUrl => Url("/ready"); }

File: clients/c/src/act_client.c
#include "act_client.h"
act_client act_client_new(const char *base_url, const char *bearer_token) {
  act_client value = {base_url, bearer_token}; return value;
}
bool act_client_health(const act_client *client) { return client != 0 && client->base_url != 0; }
